architecture/platforms/versal_acap.rst

```diff
@@ -0,0 +1,125 @@
+.. _versal_acap:
+
+#############################################
+VERSAL Adaptive Compute Acceleration Platform
+#############################################
+
+
+.. _overview :
+
+Overview
+********
+
+The OP-TEE support for the AMD/Xilinx Versal Adaptive Compute Acceleration Platform (`Versal ACAP`_) delegates most of its functionality to the `PLM firmware`_ executing in the MicroBlaze processor.
+In the case of cryptographic operations, those operations or ciphers or keys not supported by the hardware will be routed to the `Libtomcrypt`_ software based implementation.
+
+The services offered by PLM firmware are decided at build time: it is therefore important that the PLM firmware build configuration enables the services that OP-TEE will require.
+
+As an example, if OP-TEE is configured to generate a hardware unique key, it  will need access to the PLM Physical Unclonable Function and NVM services.
+
+.. note::
+   Communication between OP-TEE and the PLM uses the IPI mailbox controller being the IPI used selectable via ``CFG_VERSAL_MBOX_IPI_ID``.
+
+As described in freely available documentation for the Versal ACAP `boot-flow`_, the BootROM handles loading the PLM, while the PLM will handle loading the rest of the images including OP-TEE.
+
+A reference BIF file supporting an OP-TEE instance capable of loading the FPGA pdi can be seen below. In this example OP-TEE should be configured with ``CFG_DT=y`` and ``CFG_DT_ADDR=0x00001000``.
+If enabled the platform expects the FPGA bitstream at 0x40000000; the location is configurable using ``CFG_VERSAL_FPGA_DDR_ADDR``.
+
+.. code-block:: none
+
+	ROM_image:
+	{
+		image {
+                      { type=bootimage, file=vpl_gen_fixed.pdi }
+	              { type=bootloader, file=plm.elf }
+	              { core=psm, file=psmfw.elf }
+	        }
+	        image {
+	              id = 0x1c000000, name=apu_subsystem
+	              { type=raw, load=0x00001000, file=versal-vck190-revA-x-ebm-01-revA.dtb }
+	              { type=raw, load=0x40000000, file=fpga.pdi }
+	              { core=a72-0, exception_level=el-3, trustzone, file=bl31.elf }
+ 	              { core=a72-0, exception_level=el-2, file=u-boot.elf }
+	              { core=a72-0, exception_level=el-1, trustzone, load=0x60000000, startup=0x60000000, file=tee-raw.bin }
+	         }
+         }
+
+
+To build the boot-able image AMD/Xilinx uses the `bootgen tool`_; this tool aggregates all the different images in a single binary.
```

Comment:
bootable

Reply:
Add link to the Bootgen User Guide for users get more details and to have an idea of what the above means: https://www.xilinx.com/content/dam/xilinx/support/documents/sw_manuals/xilinx2021_1/ug1283-bootgen-user-guide.pdf